cmd/templ/generatecmd/watcher/watch.go

```diff
@@ -73,7 +73,19 @@ type RecursiveWatcher struct {
 	Events  chan fsnotify.Event
 	Errors  chan error
 	timerMu sync.Mutex
-	timers  map[string]*time.Timer
+	timers  map[timerKey]*time.Timer
+}
+
+type timerKey struct {
```

Comment:
Do you plan to extent this structure? It has the same fields as `fsnotify.Event` and you could take the event itself as key.

Reply:
My only thinking was that this protects from a future where the fsnotify event gets a new field that we don't want to debounce on. What do you think?

Reply:
I think your logic is sound there @joerdav. It reads well to me.

Reply:
@joerdav Sounds good. I was just puzzled, because other parts of the watcher uses `fsnotify.Event` for notification.